tests/test_error.py

```diff
@@ -0,0 +1,72 @@
+import paho.mqtt.client as mqtt
+import pytest
+from paho.mqtt.packettypes import PacketTypes
+
+from asyncio_mqtt.error import _CONNECT_RC_STRINGS, MqttCodeError, MqttConnectError
+
+
+@pytest.mark.parametrize(
+    "rc",
+    (
+        mqtt.MQTT_ERR_SUCCESS,
+        mqtt.MQTT_ERR_NOMEM,
+        mqtt.MQTT_ERR_PROTOCOL,
+        mqtt.MQTT_ERR_INVAL,
+        mqtt.MQTT_ERR_NO_CONN,
+        mqtt.MQTT_ERR_CONN_REFUSED,
+        mqtt.MQTT_ERR_NOT_FOUND,
+        mqtt.MQTT_ERR_CONN_LOST,
+        mqtt.MQTT_ERR_TLS,
+        mqtt.MQTT_ERR_PAYLOAD_SIZE,
+        mqtt.MQTT_ERR_NOT_SUPPORTED,
+        mqtt.MQTT_ERR_AUTH,
+        mqtt.MQTT_ERR_ACL_DENIED,
+        mqtt.MQTT_ERR_UNKNOWN,
+        mqtt.MQTT_ERR_ERRNO,
+        mqtt.MQTT_ERR_QUEUE_SIZE,
+        mqtt.MQTT_ERR_KEEPALIVE,
+        -1,
+    ),
+)
+def test_mqtt_code_error_int(rc: int) -> None:
+    assert str(MqttCodeError(rc)) == f"[code:{rc}] {mqtt.error_string(rc)}"
+
```

Comment:
Nice use of test parameterization. 👍 

These tests are a bit strict IMHO. I wouldn't call the return value of `MqttCodeError.__str__` as part of the stable API. I fear that the presence of this test make people assume that this *is* indeed part of the stable API. In any case, we don't have any plans to change it either, so I guess why not. :)